api/envoy/api/v2/core/protocol.proto

```diff
@@ -99,6 +99,14 @@ message Http1ProtocolOptions {
     message ProperCaseWords {
     }
 
+    message Custom {
+      // Custom header rewrite rules.
+      // In each rule of the map, the key is a case-insensitive header name. The value
+      // is the new header value, case-sensitive. This allows for custom header
+      // capitalization, eg: `x-my-header-key` -> `X-MY-HEADER-Key`
+      map<string, string> rules = 1;
+    }
```

Comment:
This is now hopefully obviated by https://github.com/envoyproxy/envoy/pull/15619